packages/flutter/lib/src/gestures/hit_test.dart

```diff
@@ -47,24 +47,25 @@ class HitTestResult {
   /// If the [path] argument is null, the [path] field will be initialized with
   /// and empty list.
   HitTestResult({ List<HitTestEntry> path })
-    : path = path ?? <HitTestEntry>[];
+    : _path = path ?? <HitTestEntry>[];
 
-  /// The list of [HitTestEntry] objects recorded during the hit test.
+  /// An unmodifiable list of [HitTestEntry] objects recorded during the hit test.
   ///
   /// The first entry in the path is the most specific, typically the one at
   /// the leaf of tree being hit tested. Event propagation starts with the most
   /// specific (i.e., first) entry and proceeds in order through the path.
-  final List<HitTestEntry> path;
+  List<HitTestEntry> get path => new List<HitTestEntry>.unmodifiable(_path);
```

Comment:
It's used by the testing harness to do non-trivial things.  We could expose an iterator instead, but there's no reason not to allow random access.